### ISSUE TITLE
Wrong Momentum Equation in Tensorflow's Kernel

### ISSUE THREAD

**Issue by eduardo-elizondo** [tensorflow/tensorflow#903]
I believe I found a slight error in the multiplication of the learning rate

Momentum Optimization has the form:
![brahms_cpmc_columbia_edu_publications_momentum_pdf](https://cloud.githubusercontent.com/assets/5315899/12606885/98bc9276-c496-11e5-8bfc-0b887bfaef71.png)
Where epsilon is the learning rate and p is the momentum.
Reference: Ning Qian's Momentum Paper (1999) http://brahms.cpmc.columbia.edu/publications/momentum.pdf

In tensorflow/core/kernel/training_ops.cc, lines 68 and 69 (ApplyMomentum) we find:
```c++
accum = accum * momentum() + grad;
var -= accum * lr();
```
Expanding this, we get:
```c++
var -= (accum*momentum() + grad) * lr();
// =>
var -= accum*momentum()*lr() + grad*lr();
```
Notice how the momentum variable is being multiplied by the learning rate, but this should not be the case. The correct form is:
```c++
accum = accum * momentum() + grad * lr();
var -= accum;
```
Here's my pull request fixing this: #904

**Comment by yaroslavvb** [comment]
Two forms produce equivalent descent path. Note that you can factor out lr() out of momentum in your formulation, and then write accum = lr()*(unscaled_accum); var-=lr()*unscaled_accum. So in our case the difference between our accum and your accum is an extra factor of lr(), but it doesn't affect the update. There are several slightly different variations of momentum equation out there which are equivalent for practical purposes. For instance, Polyak's original paper on momentum wrote update as x_{k+1}=x_k - \alpha_g grad + beta_k(x_k-x_{k-1})

**Comment by martinwicke** [comment]
Closing this. Reopen if the scaling of the stored value is important.

**Comment by eduardo-elizondo** [comment]
@yaroslavvb, thanks for your explanation! I was not entirely sure if scaling would eventually affect the results, but that cleared it up!
